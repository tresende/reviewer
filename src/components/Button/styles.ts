import styled, { css, DefaultTheme } from 'styled-components'
import { darken, lighten } from 'polished'
import { ButtonProps } from '.'

const WrapperModifiers = {
  outline: (theme: DefaultTheme) => css`
    background: ${theme.colors.white};
    color: ${theme.colors.primary};
  `,
  solid: (theme: DefaultTheme) => css`
    color: ${theme.colors.white};
    background: ${theme.colors.primary};
  `,
  disabled: () => css`
    &:disabled {
      cursor: not-allowed;
      filter: saturate(30%);
    }
  `
}

export const Wrapper = styled.button<ButtonProps>`
  ${({ theme, fillType, disabled }) => css`
    padding: 0.6rem 1.2rem;
    width: 100%;
    border-radius: ${theme.border.smallRadius};
    border: ${theme.border.default} ${theme.colors.primary};
    font-size: ${theme.font.sizes.large};
    box-shadow: 0px 8px 15px rgba(0, 0, 0, 0.1);
    font-weight: ${theme.font.bold};
    cursor: pointer;
    ${fillType && WrapperModifiers[fillType](theme)};
    ${disabled && WrapperModifiers.disabled()};
    ${!disabled &&
    css`
      &:hover {
        background-color: ${lighten(0.3, theme.colors.primary)};
        color: ${darken(0.2, theme.colors.primary)};
      }
    `}
  `}
`
