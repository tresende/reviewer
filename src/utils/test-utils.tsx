import { ReactElement } from 'react'
import { render, RenderOptions } from '@testing-library/react'

import { ThemeProvider } from 'styled-components'
import theme from 'styles/theme'
import { LoadingContext, LoadingContextData, LoadingContextDefaultValues } from 'hooks/use-loading'

type CustomRenderProps = {
  loadingProviderProps?: LoadingContextData
} & Omit<RenderOptions, 'queries'>

const customRender = (
  ui: ReactElement,
  { loadingProviderProps = LoadingContextDefaultValues, ...renderOptions }: CustomRenderProps = {}
) =>
  render(
    <ThemeProvider theme={theme}>
      <LoadingContext.Provider value={loadingProviderProps}>{ui}</LoadingContext.Provider>
    </ThemeProvider>,
    renderOptions
  )

export * from '@testing-library/react'
export { customRender as render }
