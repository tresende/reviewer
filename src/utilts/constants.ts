const SCORE_LABEL = ['PÃ©ssimo', 'Ruim', 'Mediano', 'Bom', 'Perfeito']
const MAX_RATING = 5
export { SCORE_LABEL, MAX_RATING }
