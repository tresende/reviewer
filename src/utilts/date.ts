export const toPTBRFormat = (date: Date | string) => {
  let toFormat = date
  if (typeof date === 'string') toFormat = new Date(date)
  const formatedDate = new Intl.DateTimeFormat('pt-br').format(toFormat)
  return formatedDate
}
