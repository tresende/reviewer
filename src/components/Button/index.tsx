import { ButtonHTMLAttributes } from 'react'

import * as S from './styles'

export type ButtonProps = {
  children: React.ReactNode
} & ButtonHTMLAttributes<HTMLButtonElement>

const Button = ({ children, ...props }: ButtonProps) => <S.Wrapper {...props}>{children}</S.Wrapper>

export default Button
