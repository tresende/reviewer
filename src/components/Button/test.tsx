import { render, screen } from 'utilts/test-utils'

import Button from '.'

describe('<Button />', () => {
  it('should render outline button by default', () => {
    const { container } = render(<Button>Buy now</Button>)

    expect(screen.getByRole('button', { name: /Buy now/i })).toHaveStyle({
      padding: '0.6rem 1.2rem',
      'font-size': '1.8rem',
      background: 'rgb(255, 255, 255)',
      color: 'rgb(250, 186, 64)'
    })

    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render solid button ', () => {
    const { container } = render(<Button fillType="solid">Buy now</Button>)

    expect(screen.getByRole('button', { name: /Buy now/i })).toHaveStyle({
      padding: '0.6rem 1.2rem',
      'font-size': '1.8rem',
      color: 'rgb(255, 255, 255)',
      background: 'rgb(250, 186, 64)'
    })

    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render a disabled Button', () => {
    const onClick = jest.fn()

    render(
      <Button onClick={onClick} disabled>
        Buy now
      </Button>
    )
    screen.getByRole('button', { name: /buy now/i }).click()
    expect(onClick).not.toHaveBeenCalled()
  })
})
