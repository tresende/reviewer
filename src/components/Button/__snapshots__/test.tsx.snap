// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should render outline button by default 1`] = `
.c0 {
  padding: 0.6rem 1.2rem;
  width: 100%;
  border-radius: 0.4rem;
  border: 1px solid #FABA40;
  font-size: 1.8rem;
  box-shadow: 0px 8px 15px rgba(0,0,0,0.1);
  font-weight: 700;
  cursor: pointer;
  background: #FFFFFF;
  color: #FABA40;
}

.c0:hover {
  background-color: #fef0d5;
  color: #cf8905;
}

<button
  class="c0"
>
  Buy now
</button>
`;

exports[`<Button /> should render solid button  1`] = `
.c0 {
  padding: 0.6rem 1.2rem;
  width: 100%;
  border-radius: 0.4rem;
  border: 1px solid #FABA40;
  font-size: 1.8rem;
  box-shadow: 0px 8px 15px rgba(0,0,0,0.1);
  font-weight: 700;
  cursor: pointer;
  color: #FFFFFF;
  background: #FABA40;
}

.c0:hover {
  background-color: #fef0d5;
  color: #cf8905;
}

<button
  class="c0"
>
  Buy now
</button>
`;
